debezium-core/src/main/java/io/debezium/pipeline/source/AbstractSnapshotChangeEventSource.java

```diff
@@ -169,6 +171,24 @@ protected abstract SnapshotResult<O> doExecute(ChangeEventSourceContext context,
     protected void complete(SnapshotContext<P, O> snapshotContext) {
```

Comment:
What is the difference between `complete` and `completed` methods?

Reply:
`complete` method is intended take actions regardless of snapshot result. `completed` method is intended for taking actions after snapshot is successful. 
In a way, `complete` is redundant. I didn't touch that method since it's used in both Oracle and SQL connectors and it's ok to use it to take the same actions in both snapshot successful and failed cases.

Reply:
@adasari I am sorry for the delay, I commented on this one already but probably forgot to hit the send button :-(.

Generally I understand what you are saying but having two methods one named complete and the other completed is very/very confusing. Could you please think about better naming for the trio of complete/completed/aborted so it is clear cut which serves to what so there is no confusion?

Reply:
I think it would be a good idea to replace complete with completed/aborted to avoid confusion. thoughts ?

Reply:
@adasari How about adding `AutoCloseable` interface and change the `complete` to `close` as it should do the resource cleanup - according to the JavaDoc. So there will be completed/aborted to act upon snapshot result and close for resource cleanup?

Reply:
Sounds good. will update the code. thanks.